LoopKit/TherapySettings.swift

```diff
@@ -0,0 +1,108 @@
+//
+//  TherapySettings.swift
+//  LoopKit
+//
+//  Created by Anna Quinlan on 7/6/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import HealthKit
+
+public struct TherapySettings: Equatable {
+    
+    public var glucoseTargetRangeSchedule: GlucoseRangeSchedule?
+
+    public var preMealTargetRange: DoubleRange?
+
+    public var legacyWorkoutTargetRange: DoubleRange?
+
+    public var maximumBasalRatePerHour: Double?
+
+    public var maximumBolus: Double?
+
+    public var suspendThreshold: GlucoseThreshold?
+    
+    public var insulinSensitivitySchedule: InsulinSensitivitySchedule?
+    
+    public var carbRatioSchedule: CarbRatioSchedule?
+    
+    public init(
+        glucoseTargetRangeSchedule: GlucoseRangeSchedule? = nil,
+        preMealTargetRange: DoubleRange? = nil,
+        legacyWorkoutTargetRange: DoubleRange? = nil,
+        maximumBasalRatePerHour: Double? = nil,
+        maximumBolus: Double? = nil,
+        suspendThreshold: GlucoseThreshold? = nil,
+        insulinSensitivitySchedule: InsulinSensitivitySchedule? = nil,
+        carbRatioSchedule: CarbRatioSchedule? = nil
+    ){
+        self.glucoseTargetRangeSchedule = glucoseTargetRangeSchedule
+        self.preMealTargetRange = preMealTargetRange
+        self.maximumBasalRatePerHour = maximumBasalRatePerHour
+        self.maximumBolus = maximumBolus
+        self.suspendThreshold = suspendThreshold
+        self.insulinSensitivitySchedule = insulinSensitivitySchedule
+        self.carbRatioSchedule = carbRatioSchedule
+    }
+}
+
+extension TherapySettings: RawRepresentable {
+    public typealias RawValue = [String: Any]
+    private static let version = 1
+
+    public init?(rawValue: RawValue) {
+        if let glucoseRangeScheduleRawValue = rawValue["glucoseTargetRangeSchedule"] as? GlucoseRangeSchedule.RawValue {
+            self.glucoseTargetRangeSchedule = GlucoseRangeSchedule(rawValue: glucoseRangeScheduleRawValue)
+
+            // Migrate the glucose range schedule override targets
+            if let overrideRangesRawValue = glucoseRangeScheduleRawValue["overrideRanges"] as? [String: DoubleRange.RawValue] {
+                if let preMealTargetRawValue = overrideRangesRawValue["preMeal"] {
+                    self.preMealTargetRange = DoubleRange(rawValue: preMealTargetRawValue)
+                }
+                if let legacyWorkoutTargetRawValue = overrideRangesRawValue["workout"] {
+                    self.legacyWorkoutTargetRange = DoubleRange(rawValue: legacyWorkoutTargetRawValue)
+                }
+            }
+        }
```

Comment:
Unless you're initializing this struct from a persisted LoopSettings struct, you can drop this migration code.